### PR TITLE
Move object test level1 merge

### PR DESCRIPTION
**Pull request by alexc19113800**
Contains progress for the MoveObject script. I've noticed that Travis have merged TestLevel1 into Merging recently but this branch was based on TestLevel1 and cannot merge automatically into Merging.

### CODE REVIEW

**Review by tg1230 — COMMENTED** [review]
I manually merged most of this into Merging because I just merged virgils progress and testlevel1 so this branch is outdated and will need to be removed. I left out MoveObject.cs however since it needs to be cleaned up some more, so focus on getting that last script cleaned up and then I'll copy it into Merging.

File: Space-Zoologist/Assets/Scripts/Plot/GridSystem.cs [hunk 1]
```diff
@@ -269,6 +269,107 @@ public void UnhighlightHomeLocations()
...
+                    pos.x += x;
+                    pos.y += y;
+                    CellGrid[pos.x, pos.y].ContainsFood = true;
+                    CellGrid[pos.x, pos.y].Food = foodSource;
+                }
+            }
+        }
+        else {
+            // Check if the whole object is in bounds
+            for (int x = -1 * radius + 1; x <= radius; x++)
```
[the remark this answers is not in the document]
Comment (alexc19113800) [reply Space-Zoologist/Assets/Scripts/Plot/GridSystem.cs:292]: This is equivalent to -(radius - 1) in other code, I updated this in the next commit to be consistent

File: Space-Zoologist/Assets/Scripts/Plot/Utils/PlacementValidation.cs [hunk 2]
```diff
@@ -35,7 +35,49 @@ public bool IsFoodPlacementValid(Vector3 mousePosition, Item selectedItem)
...
         return CheckSurroudingTiles(gridPosition, species);
     }
+    public bool IsFoodPlacementValid(Vector3 worldPosition, FoodSourceSpecies species, GameObject foodSource = null)
+    {
+        Vector3Int gridPosition = this.GridSystem.Grid.WorldToCell(worldPosition);
+        // size 1 -> rad 0, size 3 -> rad 1 ...
+        int radius = species.Size / 2;
+        Vector3Int pos;
+
+        if (species.Size % 2 == 1)
```
[the remark this answers is not in the document]
Comment (alexc19113800) [reply Space-Zoologist/Assets/Scripts/Plot/Utils/PlacementValidation.cs:46]: Updated in next commit.

File: Space-Zoologist/Assets/Scripts/UI/Store/MoveObject.cs [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    Vector3 initialPos;
+
+    private void Start()
+    {
+        tempItem = new Item();
+    }
+
+    private Vector3Int previousLocation = default;
+    // Update is called once per frame
+    void Update()
```
[the remark this answers is not in the document]
Comment (alexc19113800) [reply Space-Zoologist/Assets/Scripts/UI/Store/MoveObject.cs:31]: Updated in next commit.

File: Space-Zoologist/Assets/Scripts/UI/Store/Sections/FoodSourceStoreSection.cs [hunk 4]
```diff
@@ -54,59 +52,28 @@ public override void OnCursorPointerUp(PointerEventData eventData)
...
+            GridSystem.AddFood(mouseGridPosition, species.Size, Food);
         }
         else
         {
             //size is even: place it at cross-center (position of tile)
-            Vector3Int Temp = mouseGridPosition;
-            Temp.x += 1;
-            Temp.y += 1;
-            Vector3 FoodLocation = base.GridSystem.Grid.CellToWorld(Temp);
+            Vector3 FoodLocation = base.GridSystem.Grid.CellToWorld(Temp); //equivalent since cell and world is 1:1, but in Vector3
```
[the remark this answers is not in the document]
Comment (alexc19113800) [reply Space-Zoologist/Assets/Scripts/UI/Store/Sections/FoodSourceStoreSection.cs]: Updated in next commit.

**Comment by alexc19113800** [comment]
@tg1230 
Awaiting next review.

**Comment by tg1230** [comment]
Manually merged. Just about working, there's a small bug that I'll talk about in checkin on 5/21.

**Pull request closed without merging**